[Transformation] Add L/l and c transformations (LEFT/RIGHT, L/R and OPEN/CLOSED)

tonhuisman opened the pull request:
Just a small addition of what seems to be missing but useful.
(Changed form first using R/r to L/l as R is used for right-justify)
Added c for OPEN/CLOSED in addition to C, resolves #2351

tonhuisman committed:
[Transformation] Add R/r transformations (LEFT/RIGHT and L/R)

TD-er committed:
Merge branch 'mega' into feature/value_formattings

Budman1758 commented:
https://github.com/letscontrolit/ESPEasy/issues/2351

any chance of getting this into the mix?  Asking for a friend....  :)

tonhuisman committed:
[Transformations] Adjusted R/r to L/l and added c

tonhuisman committed:
Merge branch 'feature/value_formattings' of https://github.com/tonhuisman/ESPEasy-1 into feature/value_formattings

giig1967g reviewed (commented):
Hi,
Removing the "R" formatting command is a breaking change.
I wouldn't approve it.

tonhuisman committed:
[Justifications] Add 'C' (Capitalize, OPEN -> Open) option

tonhuisman commented:
> Removing the "R" formatting command is a breaking change.
> I wouldn't approve it.

I added it wrongly in the first commit, initially missed the existing Right-align feature, so I corrected it to L/l.

tonhuisman commented:
> any chance of getting this into the mix? Asking for a friend.... :)

I think most of your request have already been implemented (~9 months ago, according to git), so I added 'c' (OPEN/CLOSED) as a non-breaking alternative for 'C'. If you want O/C instead, you could use the 'Ln' justification for example [Plugin#State#c] for OPEN/CLOSED or [Plugin#State#c#L1] to display O/C instead, but that was already possible with the 'C' transformation 😄.

TD-er commented on src/Misc.ino:
```diff
@@ -2127,6 +2136,17 @@ void transformValue(
+                for (uint8_t i = 0; i < value.length();i++) {
+                  if (nextCapital)
+                    value[i] = toupper(value[i]);
+                  nextCapital = (value[i] == ' ' || value[i] == '-' || value[i] == '.'); // Very simple, capitalize-first-after-space/dash/period
```
I don't think the `-` is a good separator trigger for detecting words.
Maybe not used that often in English, but in Dutch (and other languages), the - is used a lot in words.  (as you should know, judging your name :) )

tonhuisman replied:
Agreed, removed it.

tonhuisman committed:
[Justifications] Added u/l for upper/lowercasing the value, removed dash-check from C justificaion

giig1967g commented:
> > Removing the "R" formatting command is a breaking change.
> > I wouldn't approve it.
> 
> I added it wrongly in the first commit, initially missed the existing Right-align feature, so I corrected it to L/l.

I understand.

tonhuisman committed:
Merge branch 'mega' into feature/value_formattings

TD-er commented:
> 
> I understand.

@giig1967g  Does that mean, you now approve of this PR?

giig1967g commented:
Hi, 
yes it looks ok to me.
